wcfsetup/install/files/lib/system/captcha/RecaptchaHandler.class.php

```diff
@@ -116,35 +119,37 @@ public function validate()
             throw new UserInputException('recaptchaString', 'false');
         }
 
-        $request = new HTTPRequest(
+        $request = new Request(
+            'GET',
             \sprintf(
-                'https://www.google.com/recaptcha/api/siteverify?secret=%s&response=%s&remoteip=%s',
-                \rawurlencode($key),
-                \rawurlencode($this->response),
-                \rawurlencode(UserUtil::getIpAddress())
-            ),
-            ['timeout' => 10]
+                'https://www.google.com/recaptcha/api/siteveify?%s',
+                \http_build_query([
+                    'secret' => $key,
+                    'response' => $this->response,
+                    'remoteip' => UserUtil::getIpAddress(),
+                ], '', '&')
+            )
         );
 
         try {
-            $request->execute();
-            $reply = $request->getReply();
-            $data = JSON::decode($reply['body']);
+            $response = $this->getHttpClient()->send($request);
+            $data = JSON::decode((string)$response->getBody());
 
             if ($data['success']) {
                 // yeah
             } else {
                 throw new UserInputException('recaptchaString', 'false');
             }
-        } catch (\Exception $e) {
-            if ($e instanceof UserInputException) {
-                throw $e;
-            }
-
+        } catch (ClientExceptionInterface $e) {
             // log error, but accept captcha
             \wcf\functions\exception\logThrowable($e);
         }
 
         WCF::getSession()->register('recaptchaDone', true);
     }
+
+    private function getHttpClient(): ClientInterface
```

Comment:
Is this method really necessary?

Reply:
Necessary is a strong word. I added it for consistency with other classes making HTTP Requests to keep the initialization of the client in an easy-to-find place.